recipes/boost/all/conanfile.py

```diff
@@ -282,6 +282,20 @@ def _has_cppstd_14_supported(self):
             # supported_cppstd: lists GCC 5 due partial support for C++14, but not enough for Boost
             return (compiler_version >= required_compiler_version) and supported_cxx14
 
+    @property
+    def _has_cppstd_20_supported(self):
+        cppstd = self.settings.compiler.get_safe("cppstd")
+        if cppstd:
+            return valid_min_cppstd(self, 20)
```

Comment:
I suspect that this isn't sufficient to exclude all invalid compilers. `valid_min_cppstd` returns `True` for gcc 8, gcc 9, gcc 10, clang 10, clang 11 clang 12 and clang 13 - all of which have partial support for C++20, but the features that cobalt relies on aren't implemented (libstdc++, this may say more about the gcc version installed on the docker images for clang than the clang versions). Unfortunately none of these compiler versions are in the test matrix, so we're not seeing this reflected in CI

Reply:
@samuel-emrys It's hard to track partial features implemented by compiler as we don't run a code to validate it. Still, if you think is something to be improved, please, open an issue to https://github.com/conan-io/conan/issues.

About Boost, we have a other checks related to the compiler version too, not only valid_min_cppstd. Now, users are suppose the be apart of their compilers, and what features they support. Here we have all those checks to avoid breaking when building, so we can inform in advance.